uwblueprint/paramedics-react
# Fixed inconsistent dates for different event pages

[Notion ticket](https://www.notion.so/uwblueprintexecs/Edit-Event-Page-Event-calendar-display-s-today-s-date-not-the-event-date-86c88f5c2a40478a830f66628bb1fedb)

## Changes
- Because we only collect month/day, time zone gets messed up when parsing from the db. `EventCreationPage` should now retrieve the date as a string from the DB (instead of parsing it into a Date object as before). 
- Date picker will select as a date but pass to DB as string in the format `yyyy-mm-dd`. 
- Fixed off by one error for month label array

## Screenshots

These three views should be the same: 

![Screen Shot 2020-11-03 at 11 57 56 PM](https://user-images.githubusercontent.com/26718994/98071004-68c5f780-1e30-11eb-8376-8627ff338272.png)
![Screen Shot 2020-11-03 at 11 58 17 PM](https://user-images.githubusercontent.com/26718994/98071019-73808c80-1e30-11eb-9398-b5082118e167.png)
![Screen Shot 2020-11-03 at 11 58 28 PM](https://user-images.githubusercontent.com/26718994/98071038-7bd8c780-1e30-11eb-9672-c02ea8d062d8.png)


## Testing
- Verify event dates are the same on home page, event page, and edit event page
- Try adding or editing an event and making sure the date you select is the date that's set. Also check in backend to make sure it looks correct in the db.

## Checklist
>[Code review doc for reference](https://www.notion.so/uwblueprintexecs/Code-Review-a21fc85b00394f488e92d9d605f6b2bc)

before opening PR

filling out PR

after opening PR
